File: src/main/webapp/scripts/controllers/shareController.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


angular.module('sbAdminApp').controller('ShareCtrl', ['$sce','$scope','ShareFactory','$rootScope',
    function ($sce, $scope, ShareFactory,$rootScope) {
        
        $scope.disabled=true;      
        $scope.compra = $rootScope.compra;//CompraSummaryFactory.get({id:$scope.compraId});
        $scope.texto = ShareFactory.get({id:$scope.compra.comId});
        $scope.getInfo = function () {
            $scope.sharePost = $sce.trustAsHtml($scope.texto.text);
        };
        $scope.shareAction = function(){
            ShareFactory.update({id:$scope.compra.comId})
        }
    }]);

angular.module('sbAdminApp').factory('ShareFactory', function ($resource) {
    return $resource('/StampUreStyle2.0/webresources/comparte/:id', {}, {
        get: {method: 'GET', params: {id: '@id'}, isArray: false},
        update: {method: 'PUT', params: {id: '@id'}}
    });
});

File: src/main/java/com/thinktanksoft/stampurestyle/component/ShareController.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.thinktanksoft.stampurestyle.component;

/**
 *
 * @author CesarAugusto
 */
public class ShareController {
    
    private ShareService shareService;
    
    public void setService(ShareService service){
        shareService = service;        
    }
    
    public String share(long id){
        return shareService.share(id);
    }
    
    public String getInfo(long id){
        return shareService.get(id);
    }
    
}
